### PR TITLE
Assert dir creation and file copy in Maven and Gradle code analysis tests

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            tl.debug(`mkdir '${dir}'`);
+            try {
+                fs.mkdirSync(dir);
+            } catch (err) {
+                throw new Error(tl.loc('LIB_MkdirFailed', directoryPath, err.message)); // Unable to create directory '{p}'. {err.message}
+            }
+        }
+    }
+
+    public static checkShell(cmd: string, continueOnError?: boolean) {
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts]: does this need to be public?

File: Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        shell.cp('-f', sourcePath, destinationPath);
+
+        this.checkShell('cp', false);
+    }
+
+    /**
+     * Create a directory at the specified path, including any folders in between.
+     * @param directoryPath Path to create.
+     */
+    public static createDirectory(directoryPath:string):void {
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts]: Is this copied from somewhere? Please add a comment with where this came from.

Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts]: Are there any unit tests that can be copied for this? There is quite a lot of code here. If not, can you add some basic unit tests?

File: Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import shell = require('shelljs');
+import path = require('path');
+import fs = require('fs');
+
+import tl = require('vsts-task-lib/task');
+
+export class FileSystemInteractions {
+    public static copyFile(sourcePath:string, destinationPath:string):void {
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts]: Add comments for public functions.

File: Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts (hunk 1 above)
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts]: No, it doesn't. Changed.

File: Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts (hunk 2 above)
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts]: In progress.

Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts]: Added ref to source.

**Pull request merged**